core_backend/tests/api/test_users.py

```diff
@@ -0,0 +1,64 @@
+from typing import AsyncGenerator
+
+import pytest
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from core_backend.app.database import get_async_session
+from core_backend.app.users.models import (
+    UserAlreadyExistsError,
+    UserNotFoundError,
+    get_user_by_token,
+    get_user_by_username,
+    save_user_to_db,
+    update_user_retrieval_key,
+)
+from core_backend.app.users.schemas import UserCreate
+from core_backend.app.utils import get_key_hash
+from core_backend.tests.api.conftest import (
+    TEST_USER_RETRIEVAL_KEY,
+    TEST_USERNAME,
+)
+
+
+class TestUsers:
+    @pytest.fixture(scope="function")
+    async def asession(self) -> AsyncGenerator[AsyncSession, None]:
+        async for session in get_async_session():
+            yield session
+
+        await session.close()
+
+    async def test_save_user_to_db(self, asession: AsyncSession) -> None:
+        user = UserCreate(username="test_username_3")
```

Comment:
Are there rules around what usernames are allowed? Would that be worth testing in
another test?

Reply:
It's a question not a suggestion!!

Reply:
Nope no rules about usernames yet!